### ISSUE TITLE
Move model_utils.jl and parameter_utils.jl to MLJBase

### ISSUE THREAD

**Issue by ablaom** [alan-turing-institute/MLJModels.jl#56]
These great tools should be available to anyone implementing the MLJ model interface.

**Comment by tlienart** [comment]
Cool! Should I open a PR with these things in MLJBase? if so do you have a preference as to where the code ends up?

**Comment by ablaom** [comment]
Sure, that'd be great. 

Could be added to MLJBase/src/utilities.jl but feel free to break this up if you want. Note that  `@set_defaults` is already there and used in MLJ/src/composites.jl. It might make sense to replace its use with the newer macro, but I would say that is quite low priority - if it ain't broke don't fix it. Still, your call.

I'm assuming you'd be happy to also make the necessary PR on MLJModels#dev to keep #dev in synch?

**Comment by tlienart** [comment]
I'll do this after #60 is merged

**Comment by ablaom** [comment]
*Todo*

[  ] Create migration PR for  MLJBase#dev (additions)

[  ] Create migration PR for MLJModels#dev (deletions and new re-export calls)

[  ] Create documentation PR on MLJ#dev for documentation of utilities in docs/src/adding_new_models_for_general_use.md

**Comment by tlienart** [comment]
opened an issue for the last task  https://github.com/alan-turing-institute/MLJ.jl/issues/241
